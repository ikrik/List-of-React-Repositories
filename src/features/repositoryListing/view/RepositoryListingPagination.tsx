import * as React from 'react'
import styled from 'styled-components'

import { Button } from '@bootstrap-styled/v4'

interface IRepositoryListingPagination {
  totalPages: number
  pageNumber: number
  disableNextPage: boolean
  disablePreviousPage: boolean
  nextPage: () => void
  previousPage: () => void
}

const PaginationWrapper = styled.div`
  width: 100%;
  padding: 1em 0;
  display: flex;
  justify-content: space-between;
`

const ButtonWrapper = styled.div`
  flex: 0 1 auto;
`

const PagesInfoWrapper = styled.div`
  flex: 0 1 auto;
  padding: 0.5em 0;
`

const RepositoryListingPagination = (props: IRepositoryListingPagination) => {
  const { totalPages, pageNumber, disablePreviousPage, disableNextPage, nextPage, previousPage } = props
  return totalPages === 0 ? (
    <></>
  ) : (
    <PaginationWrapper>
      <ButtonWrapper>
        <Button disabled={disablePreviousPage} onClick={previousPage} color='primary'>
          Previous Page
        </Button>
      </ButtonWrapper>
      <PagesInfoWrapper>
        {pageNumber}/{totalPages} pages
      </PagesInfoWrapper>
      <ButtonWrapper>
        <Button disabled={disableNextPage} onClick={nextPage} color='primary'>
          Next Page
        </Button>
      </ButtonWrapper>
    </PaginationWrapper>
  )
}

export default RepositoryListingPagination
