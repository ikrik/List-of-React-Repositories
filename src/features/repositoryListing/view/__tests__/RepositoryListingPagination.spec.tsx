import { cleanup, fireEvent, render } from '@testing-library/react'
import { mount } from 'enzyme'
import * as React from 'react'

import RepositoryListingPagination from '../RepositoryListingPagination'

describe('Manual Inputs > Trade Op main view component tests', () => {
  afterEach(cleanup)

  it('renders nothing when totalPages is zero', () => {
    const wrapper = mount(
      <RepositoryListingPagination
        totalPages={0}
        pageNumber={1}
        disableNextPage={false}
        disablePreviousPage={false}
        nextPage={jest.fn()}
        previousPage={jest.fn()}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('renders pagination when totalPage is bigger than 0', () => {
    const wrapper = mount(
      <RepositoryListingPagination
        totalPages={5}
        pageNumber={1}
        disableNextPage={false}
        disablePreviousPage={false}
        nextPage={jest.fn()}
        previousPage={jest.fn()}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('previous page is clicked', () => {
    const spyPrevious = jest.fn()

    const { getByText } = render(
      <RepositoryListingPagination
        totalPages={5}
        pageNumber={1}
        disableNextPage={false}
        disablePreviousPage={false}
        nextPage={jest.fn()}
        previousPage={spyPrevious}
      />
    )
    expect(spyPrevious).toHaveBeenCalledTimes(0)
    const previousPageButton = getByText('Previous Page')
    fireEvent.click(previousPageButton)
    expect(spyPrevious).toHaveBeenCalledTimes(1)
  })

  it('next page is clicked', () => {
    const spyNext = jest.fn()
    const spyPrevious = jest.fn()

    const { getByText } = render(
      <RepositoryListingPagination
        totalPages={5}
        pageNumber={1}
        disableNextPage={false}
        disablePreviousPage={false}
        nextPage={spyNext}
        previousPage={spyPrevious}
      />
    )
    expect(spyNext).toHaveBeenCalledTimes(0)
    const nextPageButton = getByText('Next Page')
    fireEvent.click(nextPageButton)
    expect(spyNext).toHaveBeenCalledTimes(1)
  })

  it('previous page button not clickable when disablePreviousPage is true', () => {
    const spyNext = jest.fn()
    const spyPrevious = jest.fn()

    const { getByText } = render(
      <RepositoryListingPagination
        totalPages={5}
        pageNumber={1}
        disableNextPage={false}
        disablePreviousPage={true}
        nextPage={spyNext}
        previousPage={spyPrevious}
      />
    )
    expect(spyPrevious).toHaveBeenCalledTimes(0)
    const previousPageButton = getByText('Previous Page')
    fireEvent.click(previousPageButton)
    expect(spyPrevious).toHaveBeenCalledTimes(0)
  })

  it('next page button not clickable when disableNextPage is true', () => {
    const spyNext = jest.fn()
    const spyPrevious = jest.fn()

    const { getByText } = render(
      <RepositoryListingPagination
        totalPages={5}
        pageNumber={1}
        disableNextPage={true}
        disablePreviousPage={false}
        nextPage={spyNext}
        previousPage={spyPrevious}
      />
    )
    expect(spyNext).toHaveBeenCalledTimes(0)
    const nextPageButton = getByText('Next Page')
    fireEvent.click(nextPageButton)
    expect(spyNext).toHaveBeenCalledTimes(0)
  })
})
