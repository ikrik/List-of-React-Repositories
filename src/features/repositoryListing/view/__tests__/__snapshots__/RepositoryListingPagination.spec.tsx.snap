// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manual Inputs > Trade Op main view component tests renders nothing when totalPages is zero 1`] = `
<RepositoryListingPagination
  disableNextPage={false}
  disablePreviousPage={false}
  nextPage={[MockFunction]}
  pageNumber={1}
  previousPage={[MockFunction]}
  totalPages={0}
/>
`;

exports[`Manual Inputs > Trade Op main view component tests renders pagination when totalPage is bigger than 0 1`] = `
<RepositoryListingPagination
  disableNextPage={false}
  disablePreviousPage={false}
  nextPage={[MockFunction]}
  pageNumber={1}
  previousPage={[MockFunction]}
  totalPages={5}
>
  <styled.div>
    <div
      className="sc-bdfBwQ gzcNY"
    >
      <styled.div>
        <div
          className="sc-gsTCUz hLsmvg"
        >
          <Button
            color="primary"
            disabled={false}
            focus={false}
            hover={false}
            onClick={[MockFunction]}
            tag="button"
            theme={
              Object {
                "$btn-active-box-shadow": "inset 0 3px 5px rgba(0, 0, 0, 0.125)",
                "$btn-block-spacing-y": ".5rem",
                "$btn-border-radius": ".25rem",
                "$btn-border-radius-lg": ".3rem",
                "$btn-border-radius-sm": ".2rem",
                "$btn-border-width": "1px",
                "$btn-box-shadow": "inset 0 1px 0 rgba(255, 255, 255, 0.15), 0 1px 1px rgba(0, 0, 0, 0.075)",
                "$btn-danger-bg": "#d9534f",
                "$btn-danger-border": "#d9534f",
                "$btn-danger-color": "#fff",
                "$btn-focus-box-shadow": "0 0 0 2px rgba(2, 117, 216, 0.25)",
                "$btn-info-bg": "#5bc0de",
                "$btn-info-border": "#5bc0de",
                "$btn-info-color": "#fff",
                "$btn-line-height": "1.25",
                "$btn-link-disabled-color": "#636c72",
                "$btn-padding-x": "1rem",
                "$btn-padding-x-lg": "1.5rem",
                "$btn-padding-x-sm": ".5rem",
                "$btn-padding-y": ".5rem",
                "$btn-padding-y-lg": ".75rem",
                "$btn-padding-y-sm": ".25rem",
                "$btn-primary-bg": "#0275d8",
                "$btn-primary-border": "#0275d8",
                "$btn-primary-color": "#fff",
                "$btn-secondary-bg": "#fff",
                "$btn-secondary-border": "#ccc",
                "$btn-secondary-color": "#292b2c",
                "$btn-success-bg": "#5cb85c",
                "$btn-success-border": "#5cb85c",
                "$btn-success-color": "#fff",
                "$btn-transition": "all .2s ease-in-out",
                "$btn-warning-bg": "#f0ad4e",
                "$btn-warning-border": "#f0ad4e",
                "$btn-warning-color": "#fff",
                "$enable-hover-media-query": true,
                "$enable-rounded": true,
                "$enable-shadows": false,
                "$enable-transitions": true,
              }
            }
          >
            <ButtonUnstyled
              className="Button-h8watj-0 csWZDI"
              color="primary"
              disabled={false}
              focus={false}
              hover={false}
              onClick={[MockFunction]}
              tag="button"
              theme={
                Object {
                  "$btn-active-box-shadow": "inset 0 3px 5px rgba(0, 0, 0, 0.125)",
                  "$btn-block-spacing-y": ".5rem",
                  "$btn-border-radius": ".25rem",
                  "$btn-border-radius-lg": ".3rem",
                  "$btn-border-radius-sm": ".2rem",
                  "$btn-border-width": "1px",
                  "$btn-box-shadow": "inset 0 1px 0 rgba(255, 255, 255, 0.15), 0 1px 1px rgba(0, 0, 0, 0.075)",
                  "$btn-danger-bg": "#d9534f",
                  "$btn-danger-border": "#d9534f",
                  "$btn-danger-color": "#fff",
                  "$btn-focus-box-shadow": "0 0 0 2px rgba(2, 117, 216, 0.25)",
                  "$btn-info-bg": "#5bc0de",
                  "$btn-info-border": "#5bc0de",
                  "$btn-info-color": "#fff",
                  "$btn-line-height": "1.25",
                  "$btn-link-disabled-color": "#636c72",
                  "$btn-padding-x": "1rem",
                  "$btn-padding-x-lg": "1.5rem",
                  "$btn-padding-x-sm": ".5rem",
                  "$btn-padding-y": ".5rem",
                  "$btn-padding-y-lg": ".75rem",
                  "$btn-padding-y-sm": ".25rem",
                  "$btn-primary-bg": "#0275d8",
                  "$btn-primary-border": "#0275d8",
                  "$btn-primary-color": "#fff",
                  "$btn-secondary-bg": "#fff",
                  "$btn-secondary-border": "#ccc",
                  "$btn-secondary-color": "#292b2c",
                  "$btn-success-bg": "#5cb85c",
                  "$btn-success-border": "#5cb85c",
                  "$btn-success-color": "#fff",
                  "$btn-transition": "all .2s ease-in-out",
                  "$btn-warning-bg": "#f0ad4e",
                  "$btn-warning-border": "#f0ad4e",
                  "$btn-warning-color": "#fff",
                  "$enable-hover-media-query": true,
                  "$enable-rounded": true,
                  "$enable-shadows": false,
                  "$enable-transitions": true,
                }
              }
            >
              <button
                className="Button-h8watj-0 csWZDI btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Previous Page
              </button>
            </ButtonUnstyled>
          </Button>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-dlfnbm dPRurm"
        >
          1
          /
          5
           pages
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-gsTCUz hLsmvg"
        >
          <Button
            color="primary"
            disabled={false}
            focus={false}
            hover={false}
            onClick={[MockFunction]}
            tag="button"
            theme={
              Object {
                "$btn-active-box-shadow": "inset 0 3px 5px rgba(0, 0, 0, 0.125)",
                "$btn-block-spacing-y": ".5rem",
                "$btn-border-radius": ".25rem",
                "$btn-border-radius-lg": ".3rem",
                "$btn-border-radius-sm": ".2rem",
                "$btn-border-width": "1px",
                "$btn-box-shadow": "inset 0 1px 0 rgba(255, 255, 255, 0.15), 0 1px 1px rgba(0, 0, 0, 0.075)",
                "$btn-danger-bg": "#d9534f",
                "$btn-danger-border": "#d9534f",
                "$btn-danger-color": "#fff",
                "$btn-focus-box-shadow": "0 0 0 2px rgba(2, 117, 216, 0.25)",
                "$btn-info-bg": "#5bc0de",
                "$btn-info-border": "#5bc0de",
                "$btn-info-color": "#fff",
                "$btn-line-height": "1.25",
                "$btn-link-disabled-color": "#636c72",
                "$btn-padding-x": "1rem",
                "$btn-padding-x-lg": "1.5rem",
                "$btn-padding-x-sm": ".5rem",
                "$btn-padding-y": ".5rem",
                "$btn-padding-y-lg": ".75rem",
                "$btn-padding-y-sm": ".25rem",
                "$btn-primary-bg": "#0275d8",
                "$btn-primary-border": "#0275d8",
                "$btn-primary-color": "#fff",
                "$btn-secondary-bg": "#fff",
                "$btn-secondary-border": "#ccc",
                "$btn-secondary-color": "#292b2c",
                "$btn-success-bg": "#5cb85c",
                "$btn-success-border": "#5cb85c",
                "$btn-success-color": "#fff",
                "$btn-transition": "all .2s ease-in-out",
                "$btn-warning-bg": "#f0ad4e",
                "$btn-warning-border": "#f0ad4e",
                "$btn-warning-color": "#fff",
                "$enable-hover-media-query": true,
                "$enable-rounded": true,
                "$enable-shadows": false,
                "$enable-transitions": true,
              }
            }
          >
            <ButtonUnstyled
              className="Button-h8watj-0 csWZDI"
              color="primary"
              disabled={false}
              focus={false}
              hover={false}
              onClick={[MockFunction]}
              tag="button"
              theme={
                Object {
                  "$btn-active-box-shadow": "inset 0 3px 5px rgba(0, 0, 0, 0.125)",
                  "$btn-block-spacing-y": ".5rem",
                  "$btn-border-radius": ".25rem",
                  "$btn-border-radius-lg": ".3rem",
                  "$btn-border-radius-sm": ".2rem",
                  "$btn-border-width": "1px",
                  "$btn-box-shadow": "inset 0 1px 0 rgba(255, 255, 255, 0.15), 0 1px 1px rgba(0, 0, 0, 0.075)",
                  "$btn-danger-bg": "#d9534f",
                  "$btn-danger-border": "#d9534f",
                  "$btn-danger-color": "#fff",
                  "$btn-focus-box-shadow": "0 0 0 2px rgba(2, 117, 216, 0.25)",
                  "$btn-info-bg": "#5bc0de",
                  "$btn-info-border": "#5bc0de",
                  "$btn-info-color": "#fff",
                  "$btn-line-height": "1.25",
                  "$btn-link-disabled-color": "#636c72",
                  "$btn-padding-x": "1rem",
                  "$btn-padding-x-lg": "1.5rem",
                  "$btn-padding-x-sm": ".5rem",
                  "$btn-padding-y": ".5rem",
                  "$btn-padding-y-lg": ".75rem",
                  "$btn-padding-y-sm": ".25rem",
                  "$btn-primary-bg": "#0275d8",
                  "$btn-primary-border": "#0275d8",
                  "$btn-primary-color": "#fff",
                  "$btn-secondary-bg": "#fff",
                  "$btn-secondary-border": "#ccc",
                  "$btn-secondary-color": "#292b2c",
                  "$btn-success-bg": "#5cb85c",
                  "$btn-success-border": "#5cb85c",
                  "$btn-success-color": "#fff",
                  "$btn-transition": "all .2s ease-in-out",
                  "$btn-warning-bg": "#f0ad4e",
                  "$btn-warning-border": "#f0ad4e",
                  "$btn-warning-color": "#fff",
                  "$enable-hover-media-query": true,
                  "$enable-rounded": true,
                  "$enable-shadows": false,
                  "$enable-transitions": true,
                }
              }
            >
              <button
                className="Button-h8watj-0 csWZDI btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Next Page
              </button>
            </ButtonUnstyled>
          </Button>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RepositoryListingPagination>
`;
